docs/messaging/gcp-pubsub.md

```diff
@@ -19,4 +19,42 @@ For Google Cloud Pub/Sub, the following additional attributes are defined:
 | [`messaging.gcp_pubsub.message.ordering_key`](../attributes-registry/messaging.md) | string | The ordering key for a given message. If the attribute is not present, the message does not have an ordering key. | `ordering_key` | Conditionally Required: If the message type has an ordering key set. |
 <!-- endsemconv -->
 
+## Examples
+
+### Asynchronous Batch Publish Example
+
+Given is a process P that asynchronously publishes 2 messages in a batch to a topic T on Pub/Sub.
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  direction LR
+  CA[Span Create A]
+  CB[Span Create B]
+  P[Span Publish A B] 
+  end
+  CA-. link .-P;
+  CB-. link .-P;
+
+  classDef producer fill:green
+  class P,CA,CB producer
+  classDef normal fill:green
+  class PA,PB,D1 normal
+  linkStyle 0,1 color:green,stroke:green
+```
+
+| Field or Attribute | Span Create A | Span Create B | Span Publish A B |
+|-|-|-|-|
+| Span name | `T create` | `T create` | `publish` |
+| Parent |  |  |  |
+| Links |  |  | Span Create A, Span Create B |
+| SpanKind | `PRODUCER` | `PRODUCER` | `CLIENT |
```

Comment:
I'm still trying to grok the nuances between a `PRODUCER` and `CLIENT` span. For my understanding, is this the case from [the main spec](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/messaging/messaging-spans.md#:~:text=For%20cases%20not,to%20the%20intermediary.) where our publish spans are detached from the `create` span and also is synchronous therefore it should be `CLIENT`?

Reply:
My understanding is that the `publish` span is a [client](https://opentelemetry.io/docs/concepts/signals/traces/#client) because it is an async operation AND its context is NOT injected into the message attributes.